pkg/sources/source.go

```diff
@@ -99,15 +105,22 @@ func (sp *SourceProcessor) Start(ctx context.Context) error {
 			writersMap[e.To] = bufferWriters
 		}
 	case dfv1.ISBSvcTypeJetStream:
-		// build watermark progressors
-		fetchWatermark, err = jetstream.BuildFetcher(ctx, sp.VertexInstance, natsClientPool.NextAvailableClient())
+		// build processor manager which will keep track of all the processors using heartbeat and updates their offset timelines
+		processorManagers, err = jetstream.BuildProcessorManagers(ctx, sp.VertexInstance, natsClientPool.NextAvailableClient())
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to build processor manager: %w", err)
 		}
-		toVertexWatermarkStores, err = jetstream.BuildToVertexPublisherStores(ctx, sp.VertexInstance, natsClientPool.NextAvailableClient())
+
+		// create watermark fetcher using processor managers
+		fetchWatermark = fetch.NewSourceFetcher(ctx, processorManagers[sp.VertexInstance.Vertex.Name])
+
+		// build publisher stores for to vertex
+		toVertexWatermarkStores, err = jetstream.BuildToVertexWatermarkStores(ctx, sp.VertexInstance, natsClientPool.NextAvailableClient())
 		if err != nil {
 			return err
 		}
+
+		// build publisher stores for source (we publish twice for source)
```

Comment:
could you add a comment here to explain why we need two sets of publisher stores?..